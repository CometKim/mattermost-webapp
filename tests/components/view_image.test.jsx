import React from 'react';

import {shallow} from 'enzyme';

import ViewImageModal from 'components/view_image';

describe('components/ViewImageModal', () => {
    test('should match snapshot', () => {
        const fileInfo = {
            id: '{file_id}'
        };

        const wrapper = shallow(
            <ViewImageModal
                show={true}
                onModalDismissed={jest.fn()}
                fileInfos={[fileInfo]}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
