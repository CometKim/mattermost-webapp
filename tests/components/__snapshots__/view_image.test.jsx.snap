// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ViewImageModal should match snapshot 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  className="modal-image"
  dialogClassName="modal-image"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
    modalClassName="modal-image__body"
    onClick={[Function]}
  >
    <div
      className="modal-image__wrapper"
      onClick={[Function]}
    >
      <div
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="modal-close"
          onClick={[Function]}
        />
        <div
          className="modal-image__content"
        >
          <LoadingImagePreview
            loading="Loading "
            progress={0}
          />
        </div>
        <ViewImagePopoverBar
          fileIndex={0}
          fileURL="/api/v4/files/{file_id}"
          filename=""
          onGetPublicLink={[Function]}
          show={false}
          totalFiles={1}
        />
      </div>
    </div>
  </ModalBody>
</Modal>
`;
